spec/unit/actions/collect_data_product_spec.rb

```diff
@@ -55,4 +55,26 @@
       expect(result[:params][:data_product]).to be_a GoodData::DataProduct
     end
   end
+
+  context 'when the passed data_product does not exist' do
+    let(:data_product_id) { 'non-existing-data-product' }
+    let(:params) do
```

Comment:
Identical code found in collect_data_product_spec.rb:25, collect_data_product_spec.rb:61 (mass = 72)